pkg/infrastructure/persistence/tab_repository.go

```diff
@@ -0,0 +1,106 @@
+package persistence
+
+import (
+	"context"
+
+	"github.com/iota-agency/iota-sdk/pkg/composables"
+	"github.com/iota-agency/iota-sdk/pkg/domain/entities/tab"
+	"github.com/iota-agency/iota-sdk/pkg/graphql/helpers"
+	"github.com/iota-agency/iota-sdk/pkg/infrastructure/persistence/models"
+	"github.com/iota-agency/iota-sdk/pkg/service"
+)
+
+type GormTabRepository struct{}
+
+func NewTabRepository() tab.Repository {
+	return &GormTabRepository{}
+}
+
+func (g *GormTabRepository) Count(ctx context.Context) (int64, error) {
+	tx, ok := composables.UseTx(ctx)
+	if !ok {
+		return 0, service.ErrNoTx
+	}
+	var count int64
+	if err := tx.Model(&models.Tab{}).Count(&count).Error; err != nil { //nolint:exhaustruct
+		return 0, err
+	}
+	return count, nil
+}
+
+func (g *GormTabRepository) GetAll(ctx context.Context, params *tab.FindParams) ([]*tab.Tab, error) {
+	tx, ok := composables.UseTx(ctx)
+	if !ok {
+		return nil, service.ErrNoTx
+	}
+	q, err := helpers.ApplySort(tx, params.SortBy)
+	if err != nil {
+		return nil, err
+	}
+	if params.UserID != 0 {
```

Comment:
Do not overload `GetAll` method with filtering by UserId. Instead create a separate `GetUserTabs` method

Reply:
What's wrong with filtering?

Reply:
I was lazy to write it out myself, so here's the same thing from ChatGPT:

<img width="738" alt="image" src="https://github.com/user-attachments/assets/1b28fd4c-dec2-4a1a-bd03-8b45a546ef44">

Reply:
fixed